presets/aws_sec_lake/cloudtrail_account_change/preset.yaml

```
@@ -0,0 +1,192 @@
AMP,deleted_time_dt:TIMESTAMP,email_addrs:ARRAY<STRING>,employee_uid:STRING,given_name:STRING,hire_time:TIMESTAMP,hire_time_dt:TIMESTAMP,job_title:STRING,labels:ARRAY<STRING>,last_login_time:TIMESTAMP,last_login_time_dt:TIMESTAMP,ldap_cn:STRING,ldap_dn:STRING,leave_time:TIMESTAMP,leave_time_dt:TIMESTAMP,location:STRUCT<city:STRING,continent:STRING,coordinates:ARRAY<FLOAT>,country:STRING,desc:STRING,geohash:STRING,is_on_premises:BOOLEAN,isp:STRING,lat:FLOAT,long:FLOAT,postal_code:STRING,provider:STRING,region:STRING>,manager:VARIANT,modified_time:TIMESTAMP,modified_time_dt:TIMESTAMP,office_location:STRING,surname:STRING>,name:STRING,org:STRUCT<name:STRING,ou_name:STRING,ou_uid:STRING,uid:STRING>,risk_level:STRING,risk_level_id:INT,risk_score:INT,type:STRING,type_id:INT,uid:STRING,uid_alt:STRING>,user_result STRUCT<account:STRUCT<labels:ARRAY<STRING>,name:STRING,type:STRING,type_id:INT,uid:STRING>,credential_uid:STRING,domain:STRING,email_addr:STRING,full_name:STRING,groups:ARRAY<STRUCT<desc:STRING,domain:STRING,name:STRING,privileges:ARRAY<STRING>,type:STRING,uid:STRING>>,ldap_person:STRUCT<cost_center:STRING,created_time:TIMESTAMP,created_time_dt:TIMESTAMP,deleted_time:TIMESTAMP,deleted_time_dt:TIMESTAMP,email_addrs:ARRAY<STRING>,employee_uid:STRING,given_name:STRING,hire_time:TIMESTAMP,hire_time_dt:TIMESTAMP,job_title:STRING,labels:ARRAY<STRING>,last_login_time:TIMESTAMP,last_login_time_dt:TIMESTAMP,ldap_cn:STRING,ldap_dn:STRING,leave_time:TIMESTAMP,leave_time_dt:TIMESTAMP,location:STRUCT<city:STRING,continent:STRING,coordinates:ARRAY<FLOAT>,country:STRING,desc:STRING,geohash:STRING,is_on_premises:BOOLEAN,isp:STRING,lat:FLOAT,long:FLOAT,postal_code:STRING,provider:STRING,region:STRING>,manager:VARIANT,modified_time:TIMESTAMP,modified_time_dt:TIMESTAMP,office_location:STRING,surname:STRING>,name:STRING,org:STRUCT<name:STRING,ou_name:STRING,ou_uid:STRING,uid:STRING>,risk_level:STRING,risk_level_id:INT,risk_score:INT,type:STRING,type_id:INT,uid:STRING,uid_alt:STRING>"
```

Just something to be aware of in the future. We are moving away from `foreachBatch` and so its not a problem anymore, however, I have encountered some weirdness with such large schemaHints. The batch job just silently fails, no exception, no new records written out. Moving over to streaming seems to fix it.

---

This might also account for your observation around setting format to parquet but the input file being json, as when this weird behaviour occurs, it doesn't even raise errors around wrong formats.

---

interesting - I ran each individual source more than once to rule out behavior like this :)
And they all (apart from the ones with rescued_data) successfully ingested & wrote data all the way through to gold. On the old datasource implementation that used foreachBatch.

I would not rule out that databricks silently rolled out some release in the background in between my tests and yours ...

For these presets here I think it would probably be best to directly provide a real schema file. But I think that support wasn't really implemented yet when I did this.